### ISSUE TITLE
[RFC] Namespace isolation breaks serviced actors/jobs accessibility.

### ISSUE THREAD

**Issue by jovany-wang** [ray-project/ray#19429]
### Search before asking

- [X] I had searched in the [issues](https://github.com/ray-project/ray/issues) and found no similar feature requirement.

### Description

In our Ray ecosystem, we have been building many serviced jobs to provide very easy to use services for other Ray applications in Ant.
Since all user applications and our services are both in Ray ecosystem, these serviced jobs should definitely exposed in Ray-native. That means user applications access our services in Ray actor calls instead of RPC.

![image](https://user-images.githubusercontent.com/19473861/137619989-672589bc-51b2-44cb-a8a6-87aa12ea21a6.png)

Currently, our services include state notification service, actor optimizer service and actor migration service. And we're 
 building more services.
After migrating Ray code to our internal, namespace isolation breaks the actor accessibility between applications and our user applications cannot access the services. We set namespace to an empty string as a workaround under the hook.

We raise the issue here and request for more comments on the namespace concept and isolation.

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

**Comment by jovany-wang** [comment]
CC @ericl @wuisawesome @edoakes @kfstorm @raulchen

**Comment by ericl** [comment]
Doesn't setting namespace for your services fix the issue? The breakage here was intended, the namespace is anonymous by default.

**Comment by jovany-wang** [comment]
> The breakage here was intended, the namespace is anonymous by default.

I know we introduced the breaking change intentionally. But let us review if the breaking change is reasonable enough.

> Doesn't setting namespace for your services fix the issue? 

If we set both serviced applications and user applications with the same namespace, it works.
If we set serviced applications with a fixed namespace, The named serviced actors are not accessible from the user applications.
In fact, I can give a fixed namespace to all serviced applications, but we cannot give it to all user applications.

**Comment by jovany-wang** [comment]
If we can get named actor by a given namespace, thing will be addressed.

**Comment by jovany-wang** [comment]
> my_service

It looks work to me, Thanks.
